#Parameters for the flight node

loop_rate: 18

# Disables ARM requirement
manual_testing: 0

# All kinds of limits
limit:
  nav:
    alt_margin: 1.3  # Margin (+/- target) for naviagion
    max_vs:     0.3  # Max vert speed m/s for navigation
  abort:
    altitude:   3.0   # Max allowed altitude, metres
    lost_time:  100.0  # Max time without visual nav data

# PID parameters
pid:
  # Input relative alt, output throttle setting
  altitude:
    target: 1.5   # Metres
    output:
      min:  0.0   # Throttle setting
      max:  1.0   # Throttle setting
      ramp: 2.0   # Throttle / sec
      bias: 0.5   # Throttle setting
    p:      0.09  # 0.20 without bias
    i:      0.00  # 0.06 without bias
    d:      0.00  # 0.02 without bias

  # Optical flow velocity pid
  # Input imaeg m/s ahead, output pitch forward in radians
  of_pitch:
    target: 0.0   # Metres ahead
    output:
      min:  deg(-10) # Degrees backwards
      max:  deg(+10) # Degrees forward
      ramp: deg(40)  # Degrees / sec
    p:     -0.08     # neg accel fwd / 9.81
    d:     -0.00     # accel fwd / 9.81

    filter_tconst: 0.11

  # Input metres right, output roll right in radians
  of_roll:
    target: 0.0   # Metres right
    output:
      min:  deg(-10) # Degrees left
      max:  deg(+10) # Degrees right
      ramp: deg(40)  # Degrees / sec
    p:     -0.10     # neg accel right / 9.81
    d:     -0.00     # accel right / 9.81
    
    filter_tconst: 0.06
    
    
  # Input metres ahead, output pitch forward in radians
  pitch:
    target: 0.2   # Metres ahead
    output:
      min:  deg(-4) # Degrees backwards
      max:  deg(+4) # Degrees forward
      ramp: deg(20)  # Degrees / sec
      bias: deg(0.0) # Degrees forward
    p:     -0.10     # neg accel fwd / 9.81
    d:      0.00     # accel fwd / 9.81

  # Input metres right, output roll right in radians
  roll:
    target: 0.0   # Metres right
    output:
      min:  deg(-4) # Degrees left
      max:  deg(+4) # Degrees right
      ramp: deg(20)  # Degrees / sec
      bias: deg(0.0) # Degrees right
    p:     -0.10     # neg accel right / 9.81
    d:      0.00     # accel right / 9.81

  # Input radians right, output yaw rate right rad/s
  yaw:
    target: 0.0      # rad to right
    output:
      min:  deg(-30) # Degrees backwards
      max:  deg(+30) # Degrees forward
      ramp: deg(30)  # Degrees / sec
    p:     -2.0      # neg accel right / 9.81
    d:     -1.0      # accel right / 9.81

  roll_yaw_coupling: 0.08
  lost_yaw_rate:     deg(30.0)

# Pixhawk output stream
stream:
  # id:            0  # Set stream ID
  rate:           10  # Stream rate
  # on_off:     true  # boolean
